src/main/java/com/ververica/platform/entities/Commit.java

```diff
@@ -1,18 +1,45 @@
 package com.ververica.platform.entities;
 
-import java.util.Date;
-import java.util.List;
+import java.lang.reflect.Type;
+import java.time.LocalDateTime;
+import java.util.HashMap;
+import java.util.Map;
 import lombok.AllArgsConstructor;
 import lombok.Builder;
 import lombok.Data;
 import lombok.NoArgsConstructor;
+import org.apache.flink.api.common.typeinfo.TypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInfoFactory;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeinfo.Types;
 
 @Data
 @NoArgsConstructor
 @AllArgsConstructor
 @Builder
+@TypeInfo(Commit.CommitTypeInfoFactory.class)
 public class Commit {
-  private Date timestamp;
+  private LocalDateTime commitDate;
+  private String committer;
   private String author;
-  private List<FileChanged> filesChanged;
+  private LocalDateTime authorDate;
+  private FileChanged[] filesChanged;
+
+  public static class CommitTypeInfoFactory extends TypeInfoFactory<Commit> {
```

Comment:
This might be the first time I've seen a `TypeInfoFactory` in the wild 😄

Reply:
hehe, yes - I think, I may need/want to write some blog post about the use and take this as an example